### PR TITLE
indexer: add timeout to indexing a block

### PR DESCRIPTION
**Pull request by ptrus**
Add timeout to `indexBlock` as runtime queries can block forever if the oasis-node runtime is not working.

This makes it easier to debug the case, since the indexer will report runtime queries timing out.

### CODE REVIEW

File: indexer/indexer.go [hunk 1]
```diff
@@ -61,30 +59,30 @@ func (s *Service) Health() error {
...
 	}
 
-	txs, err := s.client.GetTransactionsWithResults(s.ctx, blk.Header.Round)
+	txs, err := s.client.GetTransactionsWithResults(ctx, blk.Header.Round)
 	if err != nil {
-		return ErrGetTransactionsFailed
+		return fmt.Errorf("querying transactions with results: %w", err)
 	}
 
 	err = s.backend.Index(blk, txs, s.blockGasLimit)
```
Comment (ptrus) [inline indexer/indexer.go:84]: Would be great if also this method also accepted a context, instead of using an internal one, but this fix will be done separate PR as it requires a bit of a bigger refactor.

**Review by kostko — APPROVED** [review]
